stack-z-slices/src/main.py

```diff
@@ -0,0 +1,129 @@
+                                                                                                                                                          y,
+                                                                                                                                                          c,
+                                                                                                                                                          t),
+                                                                                                                                                          shell=True,
+                                                                                                                                                          stderr=subprocess.DEVNULL))
+                        process_timer.append(time.time())
+            else:
+                ps = [p for p in files[t][c].keys()]
+                ps.sort()
+                for p in ps:
+                    if len(processes) >= multiprocessing.cpu_count()-1:
+                        free_process = False
+                        while not free_process:
+                            for process in range(len(processes)):
+                                if processes[process].poll() is not None:
+                                    free_process = process
+                        pnum += 1
+                        logger.info("Finished process {} of {} in {}s!".format(pnum,len(ts)*len(cs)*len(ps),time.time() - process_timer[free_process]))
+                        del processes[free_process]
+                        del process_timer[free_process]
+                    processes.append(subprocess.Popen("python3 merge_layers.py --inpDir {} --outDir {} --regex {} --P {} --C {} --T {}".format(input_dir,
+                                                                                                                                               output_dir,
+                                                                                                                                               file_pattern,
+                                                                                                                                               p,
+                                                                                                                                               c,
+                                                                                                                                               t),
+                                                                                                                                               shell=True,
+                                                                                                                                               stderr=subprocess.DEVNULL))
```

Comment:
So, I know this is pretty self-explanatory in terms of what its doing.... but it'd be good if you added a few more comments to describe what each for loop is for/doing.